### PR TITLE
*: merge statement buffer when BatchGetValues

### PR DESCRIPTION
**Pull request by jackysp**
### What problem does this PR solve? 
The BatchGetValues may be used several times in one statement, but it does not contain the row add in the statement itself, which will cause some keys did not check.

### What is changed and how it works?
Add an interface `BatchGet` to get the statement and the transaction key/values, and merge them with the storage BatchGet results.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change

Side effects

 - Possible performance regression
 - Increased code complexity

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

**Comment by crazycs520** [comment]
@tiancaiamao @winkyao PTAL

File: session/txn.go [hunk 1]
```diff
@@ -213,6 +213,36 @@ func (st *TxnState) Get(k kv.Key) ([]byte, error) {
...
+	for i, key := range keys {
+		val, err := st.buf.Get(key)
+		if kv.IsErrNotFound(err) {
+			shrinkKeys = append(shrinkKeys, key)
+			continue
+		}
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		if len(val) != 0 {
```
Comment (tiancaiamao) [inline session/txn.go]: if len(val) == 0, it means the key-value is deleted.
This will not reflect in the final result?

Comment (jackysp) [reply session/txn.go]: I think the deleted key should not be in the final result. It should be the same as the snapshot.BatchGet() from the kv storage.

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**